include/CLI/TypeTools.hpp

```diff
@@ -343,21 +414,112 @@ bool lexical_cast(std::string input, T &output) {
     return true;
 }
 
-/// Non-string parsable
+/// assignable from double or int
 template <typename T,
           enable_if_t<!std::is_floating_point<T>::value && !std::is_integral<T>::value &&
-                          !std::is_assignable<T &, std::string>::value && !std::is_enum<T>::value,
+                          !std::is_assignable<T &, std::string>::value &&
+                          !std::is_constructible<T, std::string>::value && !std::is_enum<T>::value &&
+                          is_direct_constructible<T, double>::value && is_direct_constructible<T, int>::value,
```

Comment:
TODO (not necessarily in this PR): I think we can make combinations of these as traits and use those; this is getting very long and hard to parse by eye.

Reply:
I made an attempt at this in PR #286